src/main/scala/scorex/core/network/peer/PeerManager.scala

```diff
@@ -65,6 +69,9 @@ class PeerManager(settings: ScorexSettings, scorexContext: ScorexContext) extend
     case RemovePeer(address) =>
       log.info(s"$address removed from peers database")
       peerDatabase.remove(address)
+      if (peerDatabaseHasOnlyKnownPeersFromConfig()) {
+        sender() ! EmptyPeerDatabase
```

Comment:
is really needed to fire this EmptyPeerDatabase message?
Maybe is enough to fire direclty GetNewPeers, unless there is a reason to go always through the NetworkController
also: instead of firing the update when the new peers is empty maybe is worth to "ping" the seeders every a reasonable amount of time (like 30 minutes?). in this way we don't have to restart a node if the dns configuration is changed. Is it possible to set Akka to fire a message every x minutes?

Reply:
This sounds like one of the options that we took into consideration: if we don't receive a new block in 15 minutes, then we need new peers because we're not connected to any forger node. The solution to check the internal database is simpler and we don't have to schedule any jobs.
Regarding passing through the NetworkController, it's more a pattern that is implementing scorex. So the peer manager, which is just a storage, detects that we don't have any more peer and tells it to the network controller, who's responsible to coordinate the different actors.

Reply:
after today discussion let'ls discard my comment (in reality the new peer will be broadcasted to the others anyway when it first tries to connect to another one).
Just one final note: I would change the message name from EmptyPeerDatabase to PeerDatabaseIsEmpty  (more clear it notifies a state)